File: 1646-kth-missing-positive-number/1646-kth-missing-positive-number.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        start,end,ans=0,len(arr)-1,len(arr)

        while start <= end:
            mid = start + (end-start)//2

            if(arr[mid]-mid-1>=k):
                ans= mid
                end=mid-1
            else:
                start=mid+1
        return ans+k

File: 1646-kth-missing-positive-number/1646-kth-missing-positive-number.go
func findKthPositive(arr []int, k int) int {
    start, end, ans := 0, len(arr)-1, len(arr)

    for start <= end {
        mid := start + (end - start) / 2

        if arr[mid] - mid - 1 >= k {
            ans = mid
            end = mid - 1
        } else {
            start = mid + 1
        }
    }
    return ans + k
}
